pkg/trace/config/subscriber.go

```diff
@@ -0,0 +1,116 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/config/remote/service"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Subscriber subscribes for configuration updates from agent-core.
+// Provides to trace-agent clients the latest available configurations.
+type Subscriber struct {
+	mu sync.RWMutex
+	// configs is the latest debugging configurations available
+	configs        pbgo.ConfigResponse
+	once           sync.Once
+	stopSubscriber context.CancelFunc
+}
+
+// NewSubscriber returns a new configuration store
+func NewSubscriber() *Subscriber {
+	return &Subscriber{}
+}
+
+// Get returns the latest available configurations
+func (s *Subscriber) Get(req *pbgo.GetConfigsRequest) (*pbgo.ConfigResponse, error) {
+	if req.Product != pbgo.Product_LIVE_DEBUGGING {
+		return nil, errors.New("not allowed")
+	}
+	s.once.Do(s.subscribe)
+	// No new configurations available in store
+	if req.CurrentConfigProductVersion >= s.getCurrentVersion() {
```

Comment:
If I understand correctly, `s.configs` is only set when a config was received through gRPC and can therefore be nil when `s.getCurrentVersion` is called.

Reply:
If no config was received through grpc, `s.configs` is the default struct  `pbgo.ConfigResponse` so `s.configs.ConfigDelegatedTargetVersion` exists and is set to 0.
Adding a test case to cover this case

Reply:
Actually it's already tested here: https://github.com/DataDog/datadog-agent/blob/79c25fda60664dbe07cd97e3e5d5a99ce34edbdb/pkg/trace/config/subscriber_test.go#L87-L92